### PR TITLE
dimming/smartthings

### PR DESCRIPTION
**Pull request by EllenPorter**

### CODE REVIEW

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js [hunk 1]
```diff
@@ -156,10 +156,53 @@ function defaultValueIfEmpty(property, defaultValue) {
...
 
 /**
+ * Given a SmartThings provider formatted Schema, calculate the
+ * desired brightness as a percentage of the current setting
+ */
+function getDesiredBrightnessState(providerSchema, resourceSchema) {
+
+    var result = {};
+
+    if(providerSchema.attributes.hasOwnProperty('level')) {
```
Comment (rgruian) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Minor: if( vs if (
Formatting consistency

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js [hunk 2]
```diff
@@ -156,10 +156,53 @@ function defaultValueIfEmpty(property, defaultValue) {
...
+        var level = providerSchema.attributes.level;
+        var desired_level = level;
+
+        if (resourceSchema.hasOwnProperty('dimmingSetPercentage')) {
+
+            desired_level = resourceSchema.dimmingSetPercentage;
+
+        } else if (resourceSchema.hasOwnProperty('dimmingIncrementPercentage')) {
+
+            var incrementBy = resourceSchema.dimmingIncrementPercentage;
```
Comment (rgruian) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Minor: do you really need the local incrementBy?

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js [hunk 3]
```diff
@@ -156,10 +156,53 @@ function defaultValueIfEmpty(property, defaultValue) {
...
+            desired_level = level+incrementBy;
+
+        } else if (resourceSchema.hasOwnProperty('dimmingDecrementPercentage')) {
+
+            var decrementBy = resourceSchema.dimmingDecrementPercentage;
```
Comment (rgruian) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Minor: same question/feedback here. The local doesn't really seem to be needed

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js (hunk 1 above)
Comment (EllenPorter) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Fixed.

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js (hunk 2 above)
Comment (EllenPorter) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Nope, removed it.

File: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js (hunk 3 above)
Comment (EllenPorter) [inline org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js]: Removed this one as well.

**Pull request merged**